### PR TITLE
[CNDIT-538] Added CLI endpoints to interact with DI Service

### PR DESCRIPTION
**Pull request by rshanmugam-enquizit**
### Notes
This PR is the first commit for the DataIngestion CLI. The main functionality of this CLI is to interact with the endpoints for the DataIngestion Service. This will replace all the testing that we are doing through Postman and also help the support team members to onboard users to DI Service.

A document will be added later for the users and also for the developers on how to use this CLI to interact with the DI Service. This CLI is created with a framework called `picocli` and can be extended further when we add more endpoints to the service.

### JIRA
- https://cdc-nbs.atlassian.net/browse/CNDIT-538

### Testing
- Tested all three Subcommands and verified it is pushing data into the database
- Unit tests added and see the coverage results below
Screenshot 2023-07-24 at 5 15 59 PM

### CODE REVIEW

**Comment by ndduc01** [comment]
Can you pump the test coverage to 90%, if anything needed to be exclude from the test. Perhaps, you can added those to read me or a note somewhere.

File: src/main/java/gov/cdc/dataingestion/commands/InjectHL7.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    AuthModel authModel = new AuthModel();
+    AuthUtil authUtil = new AuthUtil();
+
+
+    @Override
+    public void run() {
+        if(adminUser != null && adminPassword != null && hl7FilePath != null) {
+            if(!adminUser.isEmpty() && adminPassword.length > 0) {
+                String serviceEndpoint = "https://dataingestion.datateam-cdc-nbs.eqsandbox.com/api/reports";
```
Comment (ndduc01) [inline src/main/java/gov/cdc/dataingestion/commands/InjectHL7.java:33]: can this be configurable? maybe in yaml or properties

File: src/main/java/gov/cdc/dataingestion/commands/RegisterUser.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    AuthModel authModel = new AuthModel();
+    AuthUtil authUtil = new AuthUtil();
+
+
+    @Override
+    public void run() {
+        if(username != null && password != null && adminUser != null && adminPassword != null) {
+            if(!username.isEmpty() && password.length > 0 && !adminUser.isEmpty() && adminPassword.length > 0) {
+                String serviceEndpoint = "https://dataingestion.datateam-cdc-nbs.eqsandbox.com/registration?username="
```
Comment (ndduc01) [inline src/main/java/gov/cdc/dataingestion/commands/RegisterUser.java:32]: same

File: src/main/java/gov/cdc/dataingestion/commands/TokenGenerator.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    char[] adminPassword;
+
+    AuthModel authModel = new AuthModel();
+    AuthUtil authUtil = new AuthUtil();
+
+    @Override
+    public void run() {
+        if(adminUser != null && adminPassword != null) {
+            if(!adminUser.isEmpty() && adminPassword.length > 0) {
+                String serviceEndpoint = "https://dataingestion.datateam-cdc-nbs.eqsandbox.com/token";
```
Comment (ndduc01) [inline src/main/java/gov/cdc/dataingestion/commands/TokenGenerator.java:23]: same

File: src/test/java/gov/cdc/dataingestion/commands/InjectHL7Test.java [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+        injectHL7.run();
+
+        ArgumentCaptor<AuthModel> authModelCaptor = ArgumentCaptor.forClass(AuthModel.class);
+        verify(authUtilMock).getResponseFromDIService(authModelCaptor.capture());
+
+        String expectedOutput = "Dummy_UUID";
+        assertEquals("adminUser", authModelCaptor.getValue().getAdminUser());
+        assertArrayEquals("adminPassword".toCharArray(), authModelCaptor.getValue().getAdminPassword());
+//        assertEquals("MSH|^~\\&|SIMHOSP|SFAC|RAPP|RFAC|20200508130643||ADT^A01|5|T|2.3|||AL||44|ASCII\n" +
```
Comment (ndduc01) [inline src/test/java/gov/cdc/dataingestion/commands/InjectHL7Test.java:66]: remove if no longer needed

File: src/test/java/gov/cdc/dataingestion/commands/RegisterUserTest.java [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.*;
+
+class RegisterUserTest {
+    private final ByteArrayOutputStream outStream = new ByteArrayOutputStream();
+    private final ByteArrayOutputStream errStream = new ByteArrayOutputStream();
+    @Mock
+    private AuthUtil authUtilMock;
+    private RegisterUser registerUser;
+    // private String serviceEndpoint = "";
```
Comment (ndduc01) [inline src/test/java/gov/cdc/dataingestion/commands/RegisterUserTest.java:25]: remove if no longer needed

File: src/test/java/gov/cdc/dataingestion/commands/RegisterUserTest.java [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+        System.setOut(new PrintStream(outStream));
+        System.setErr(new PrintStream(errStream));
+        registerUser = new RegisterUser();
+        registerUser.authUtil = authUtilMock;
+        registerUser.authModel = new AuthModel();
+        registerUser.authModel.setServiceEndpoint("https://dataingestion.datateam-cdc-nbs.eqsandbox.com/registration?username=testUser&password=testPassword");
```
Comment (ndduc01) [inline src/test/java/gov/cdc/dataingestion/commands/RegisterUserTest.java:37]: same with other comments

File: src/test/java/gov/cdc/dataingestion/commands/TokenGeneratorTest.java [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+        System.setOut(new PrintStream(outStream));
+        System.setErr(new PrintStream(errStream));
+        tokenGenerator = new TokenGenerator();
+        tokenGenerator.authUtil = authUtilMock;
+        tokenGenerator.authModel = new AuthModel();
+        tokenGenerator.authModel.setServiceEndpoint("https://dataingestion.datateam-cdc-nbs.eqsandbox.com/token");
```
Comment (ndduc01) [inline src/test/java/gov/cdc/dataingestion/commands/TokenGeneratorTest.java:34]: same

**Review by ndduc01 — COMMENTED** [review]
Several small comments

File: src/main/java/gov/cdc/dataingestion/commands/RegisterUser.java (hunk 2 above)
Comment (rshanmugam-enquizit) [reply src/main/java/gov/cdc/dataingestion/commands/RegisterUser.java:32]: updated

File: src/main/java/gov/cdc/dataingestion/commands/TokenGenerator.java (hunk 3 above)
Comment (rshanmugam-enquizit) [reply src/main/java/gov/cdc/dataingestion/commands/TokenGenerator.java:23]: updated

File: src/test/java/gov/cdc/dataingestion/commands/InjectHL7Test.java (hunk 4 above)
Comment (rshanmugam-enquizit) [reply src/test/java/gov/cdc/dataingestion/commands/InjectHL7Test.java]: done

File: src/test/java/gov/cdc/dataingestion/commands/RegisterUserTest.java (hunk 6 above)
Comment (rshanmugam-enquizit) [reply src/test/java/gov/cdc/dataingestion/commands/RegisterUserTest.java]: done

File: src/test/java/gov/cdc/dataingestion/commands/TokenGeneratorTest.java (hunk 7 above)
Comment (rshanmugam-enquizit) [reply src/test/java/gov/cdc/dataingestion/commands/TokenGeneratorTest.java]: done

File: src/test/java/gov/cdc/dataingestion/commands/RegisterUserTest.java (hunk 5 above)
Comment (rshanmugam-enquizit) [reply src/test/java/gov/cdc/dataingestion/commands/RegisterUserTest.java]: done

**Comment by ndduc01** [comment]
looks good to me, maybe wait for Stefan, he probably has more input

**Review by ndduc01 — APPROVED** [review]

**Review by sdeusch — APPROVED** [review]

**Comment by sdeusch** [comment]
That's be a good demo next time and for monthly team CDC demo!

**Pull request merged**